datahub-graphql-core/src/main/resources/entity.graphql

```diff
@@ -5121,4 +5177,149 @@ type SubTypes {
     The sub-types that this entity implements. e.g. Datasets that are views will implement the "view" subtype
    """
    typeNames: [String!]
+}
+
+"""
+A domain, or a logical grouping of Metadata Entities
+"""
+type Domain implements Entity {
+    """
+    The primary key of the domain
+    """
+    urn: String!
+
+    """
+    A standard Entity Type
+    """
+    type: EntityType!
+
+    """
+    Id of the domain
+    """
+    id: String!
+
+    """
+    Properties about a domain
+    """
+    properties: DomainProperties
+
+    """
+    Ownership metadata of the dataset
+    """
+    ownership: Ownership
+
+    """
+    References to internal resources related to the dataset
+    """
+    institutionalMemory: InstitutionalMemory
+
+    """
+    Children entities inside of the Domain
+    """
+    entities(input: DomainEntitiesInput): SearchResults
+
+    """
+    Edges extending from this entity
+    """
+    relationships(input: RelationshipsInput!): EntityRelationshipsResult
+}
+
+"""
+Properties about a domain
+"""
+type DomainProperties {
+  """
+  Display name of the domain
+  """
+  name: String!
+
+  """
+  Description of the Domain
+  """
+  description: String
+}
+
+"""
+Input required to fetch the entities inside of a Domain.
+"""
+input DomainEntitiesInput {
+  """
+  Optional query filter for particular entities inside the domain
+  """
+  query: String
+
+  """
+  The offset of the result set
+  """
+  start: Int
+
+  """
+  The number of entities to include in result set
+  """
+  count: Int
+
+  """
+  Optional Facet filters to apply to the result set
+  """
+  filters: [FacetFilterInput!]
+}
```

Comment:
why cant we solve this via the `relationships` property? now that i think about it, this really shouldn't be a search. Isn't this just simply a graph query?